src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

```diff
@@ -1665,6 +1665,43 @@ void ReconstructionEngine_sequentialSfM::getTracksToTriangulate(const std::set<I
   }
 }
 
+namespace {
+
+struct ObservationData
+{
+  std::shared_ptr<camera::Pinhole> cam;
+  Pose3 pose;
+  Mat34 P;
+  Vec2 x;
+  Vec2 xUd;
+};
+
+ObservationData getObservationData(const SfMData& scene,
+                                   feature::FeaturesPerView* featuresPerView,
+                                   IndexT viewId, const track::Track& track)
+{
+  const View* view = scene.getViews().at(viewId).get();
+
+  std::shared_ptr<camera::IntrinsicBase> cam = scene.getIntrinsics().at(view->getIntrinsicId());
+  std::shared_ptr<camera::Pinhole> camPinHole = std::dynamic_pointer_cast<camera::Pinhole>(cam);
+
+  if (!camPinHole) {
+    ALICEVISION_LOG_ERROR("Camera is not pinhole in triangulate_multiViewsLORANSAC");
+    return {nullptr, nullptr, {}, {}, {}, {}};
```

Comment:
```suggestion
    return {nullptr, {}, {}, {}, {}};
```
as the raw pointer has been removed.

I was also wondering if it'd be worth it to have a method `isEmpty(){return cam == nullptr}` in `ObservationData` so that it is easier to test the result. What do you think?

Reply:
Fixed now. I also changed indentation of new code.